### PR TITLE
kubelet manages /etc/hosts file - cherrypick to release-1.1

### PR DESCRIPTION
**Pull request by ArtfulCoder**
cherry pick of #16052 and #16174

### CODE REVIEW

File: pkg/kubelet/dockertools/manager.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/prober"
 	"k8s.io/kubernetes/pkg/kubelet/qos"
 	kubeletTypes "k8s.io/kubernetes/pkg/kubelet/types"
+	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
```
Comment (thockin) [inline pkg/kubelet/dockertools/manager.go]: this is a dup of teh line before

Comment (thockin) [inline pkg/kubelet/dockertools/manager.go]: if it is this way in head, we can go ahead with this but we should fix head ASAP

Comment (ArtfulCoder) [inline pkg/kubelet/dockertools/manager.go]: "kubetypes" is used in master branch.
I can remove the added import.

The other issue I faced while creating the cherry-pick was that, on master branch, we now have HostNetwork field within SecurityContext within Spec within Pod type.

But the release branch does not have SecurityContext in Spec.
HostNetwork field is present in the Spec as a top level field.

So, I had to make custom changes around that.

Let me know your thoughts.
I assumed that we don't want to pull in that Spec change.

Comment (thockin) [inline pkg/kubelet/dockertools/manager.go]: Patching it up is fine.  If master has a dup import, we should fix that.

Comment (dchen1107) [inline pkg/kubelet/dockertools/manager.go]: Master doesn't have a dup import. The initial alias used here is kubeletTypes, which doesn't match the naming convention for go package. We fixed it later. I believe the 1.1 was cut between, and such cleanup code wasn't cherry-picked. 

Should we fix this in 1.1 branch by cherry-pick the cleanup code too?

Comment (thockin) [inline pkg/kubelet/dockertools/manager.go]: I can't find anything that says this is a problem, and I don't think we
should be doing manual edits on a branch unless we HAVE TO.

Comment (dchen1107) [inline pkg/kubelet/dockertools/manager.go]: https://github.com/kubernetes/kubernetes/issues/15319

Comment (ArtfulCoder) [inline pkg/kubelet/dockertools/manager.go]: for now, I removed the additional import, else I would end up doing more changes than necessary..

**Comment by dchen1107** [comment]
LGTM

**Pull request merged**